lib/core/src/Cardano/Wallet/DB/SqliteTypes.hs

```diff
@@ -0,0 +1,246 @@
+    ( Generic )
+import Web.HttpApiData
+import Web.PathPieces
+
+import qualified Data.Text as T
+
+import Cardano.Wallet.Primitive.Types
+    ( Coin (..)
+    , Direction (..)
+    , Hash (..)
+    , SlotId (..)
+    , TxStatus (..)
+    , WalletId (..)
+    , WalletState (..)
+    , flatSlot
+    , fromFlatSlot
+    , isValidCoin
+    )
+
+----------------------------------------------------------------------------
+-- Helper functions
+
+parseUrlPieceFromText :: FromText a => Text -> Either Text a
+parseUrlPieceFromText = first (T.pack . getTextDecodingError) . fromText
+
+aesonFromText :: FromText a => String -> Value -> Parser a
+aesonFromText what = withText what $ either (fail . show) pure . fromText
+
+----------------------------------------------------------------------------
+-- Direction
+
+instance PersistField Direction where
+    toPersistValue = toPersistValue . directionToBool
+    fromPersistValue pv = do
+        let err = T.pack $ "not a valid value: " <> show pv
+        bimap (const err) directionFromBool (fromPersistValue pv)
+
+instance PersistFieldSql Direction where
+    sqlType _ = sqlType (Proxy @Bool)
+
+directionToBool :: Direction -> Bool
+directionToBool Incoming = True
+directionToBool Outgoing = False
+
+directionFromBool :: Bool -> Direction
+directionFromBool True = Incoming
+directionFromBool False = Outgoing
+
+----------------------------------------------------------------------------
+-- WalletId
+
+instance PersistField WalletId where
+    toPersistValue =
+        toPersistValue . toText
+    fromPersistValue pv = do
+        a <- fromText <$> fromPersistValue pv
+        case a of
+             Left _     ->
+                 Left . T.pack $ "not a valid value: " <> show pv
+             Right txid ->
+                 pure txid
+
+instance PersistFieldSql WalletId where
+    sqlType _ = sqlType (Proxy @Text)
+
+instance Read WalletId where
+    readsPrec _ = error "readsPrec stub needed for persistent"
+
+instance ToHttpApiData WalletId where
+    toUrlPiece = toText
+
+instance FromHttpApiData WalletId where
+    parseUrlPiece = parseUrlPieceFromText
+
+instance ToJSON WalletId where
+    toJSON = String . toText
+
+instance FromJSON WalletId where
+    parseJSON = aesonFromText "WalletId"
+
+instance PathPiece WalletId where
+    fromPathPiece = fromTextMaybe
+    toPathPiece = toText
+
+
+----------------------------------------------------------------------------
+-- AddressScheme
+
+data AddressScheme = Sequential | Random deriving (Show, Eq, Generic)
+
+instance PersistField AddressScheme where
+    toPersistValue = toPersistValue . schemeToBool
+    fromPersistValue pv = do
+        let err = T.pack $ "not a valid value: " <> show pv
+        bimap (const err) schemeFromBool (fromPersistValue pv)
+
+instance PersistFieldSql AddressScheme where
+    sqlType _ = sqlType (Proxy @Bool)
+
+schemeToBool :: AddressScheme -> Bool
+schemeToBool Sequential = True
+schemeToBool Random = False
```

Comment:
Perhaps this can be renamed to `isSchemeSequential`. (Provide an equivalent `isSchemeRandom` if desired.)

Reply:
Hmmm... Just seeing this bit from the PR, but I don't think a `Bool` is a wise choice for this one... We will likely introduce more schemes in the future (special ones for exchanges), and serializing this to `Bool` might bite us later. Also, we already have a _3rd_ scheme that we use in the benchmarks `Any`, and this will need to work with the DB as well...

Reply:
It's not used yet. We can turn it into an enum later.

Reply:
changed to `data AddressScheme = Sequential | Random | Any`